.github/workflows/lint.yml

```diff
@@ -47,16 +42,8 @@ jobs:
     steps:
       - uses: actions/checkout@v1
 
-      - uses: actions/setup-python@v2
```

Comment:
Why did you delete `setup-python` here and not elsewhere?

Reply:
And one of the key reasons I'm asking is it means nothing is being cached in this step.

Reply:
> Why did you delete setup-python here and not elsewhere?

The other steps require a specific python version to match the one used in the noxfile. This step doesn't so it just uses the already installed python that nox is running with (Python 3.8).

> And one of the key reasons I'm asking is it means nothing is being cached in this step.

Regarding cache, I don't feel this project will benefit much (quite the opposite in fact):
- cache eviction policy [states 7 days retention](https://docs.github.com/en/actions/using-workflows/caching-dependencies-to-speed-up-workflows#usage-limits-and-eviction-policy): the activity I see on the project suggests that most often than not, cache will not exist and be recreated thus wasting time rather than saving it.
- If it had more activity (or a cron task to ensure no eviction), then I still feel it would not be beneficial in the current state of things and my understanding of pip cache / GHA cache: [GHA cache is not self-updating and relies only a key to create/restore cache](https://github.com/actions/setup-python#caching-packages-dependencies). Given there's no `requirements.txt` in this repo, the key is always the same and assuming there's no eviction, you always get the same cache created at one point in time. What this means given my understanding of pip cache, is that in a near future, this will happen:
  - pip will check if its cache is up-to-date for a given package (none are pinned here) and it won't be.
  - we're back to the previous state of things except we will have added some overhead for cache management.